config-service/src/main/java/org/hypertrace/config/service/ConfigServicesFactory.java

```diff
@@ -30,14 +32,17 @@
         ManagedChannelBuilder.forAddress("localhost", port).usePlaintext().build();
     lifecycle.shutdownComplete().thenRun(configChannel::shutdown);
 
+    ConfigChangeEventGenerator configChangeEventGenerator =
+        ConfigChangeEventGeneratorFactory.getInstance().createConfigChangeEventGenerator(config);
+
     return List.of(
         new ConfigServiceGrpcImpl(configStore),
         new SpacesConfigServiceImpl(configChannel),
         new LabelsConfigServiceImpl(configChannel, config),
-        new LabelApplicationRuleConfigServiceImpl(configChannel),
-        new EventConditionConfigServiceImpl(configChannel),
-        new NotificationRuleConfigServiceImpl(configChannel),
-        new NotificationChannelConfigServiceImpl(configChannel));
+        new LabelApplicationRuleConfigServiceImpl(configChannel, configChangeEventGenerator),
```

Comment:
Not sure if any of these are using guice (assuming not due to the explicit construction), but exposing a guice module for the generator would make adding it to guice-enabled services less invasive.